service/k8sclient/k8client.go

```diff
@@ -0,0 +1,273 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package k8sclient provides client for kubernetes.
+package k8sclient
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/AlekSi/pointer"
+	_ "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1" // It'll be implemented later.
+	pxc "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	"github.com/pkg/errors"
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/percona-platform/dbaas-controller/service/k8sclient/kubectl"
+	"github.com/percona-platform/dbaas-controller/utils/logger"
+)
+
+// ClusterKind is a kind of a cluster.
+type ClusterKind string
+
+const perconaXtraDBClusterKind = ClusterKind("percona-xtradb-cluster")
+
+const (
+	pxcBackupImage       = "percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0-backup"
+	pxcImage             = "percona/percona-xtradb-cluster-operator:1.4.0-pxc8.0"
+	pxcBackupStorageName = "test-backup-storage"
+)
+
+// XtraDBParams contains all parameters required to create or update Percona XtraDB cluster.
+type XtraDBParams struct {
+	Name string
+	Size int32
+}
+
+// Cluster contains information related to cluster.
+type Cluster struct {
+	Name   string
+	Kind   ClusterKind
+	Size   int32
+	Status string // FIXME
+}
+
+// K8Client is a client for Kubernetes.
+type K8Client struct {
+	kubeCtl *kubectl.KubeCtl
+}
+
+// NewK8Client returns new K8Client object.
+func NewK8Client(logger logger.Logger) *K8Client {
+	return &K8Client{
+		kubeCtl: kubectl.NewKubeCtl(logger),
+	}
+}
+
+// Cleanup removes temporary files created by that object.
+func (c *K8Client) Cleanup() {
+	c.kubeCtl.Cleanup()
+}
+
+// ListClusters returns list of clusters and their statuses.
+func (c *K8Client) ListClusters(ctx context.Context) ([]Cluster, error) {
```

Comment:
There should two method for two types of clusters – they have different properties like ProxySQL settings for PXC